File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
function square_of_sum(n)
    sum = 0
    for number in collect(1:n)
        sum += number
    end
    sum *= sum
    return sum
end

"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
    sum = 0
    for number in collect(1:n)
        sum += number * number
    end
    return sum
end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
function difference(n)
    return square_of_sum(n) - sum_of_squares(n)
end

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

    int square_of_sum(int number) {
        int result{};
        for ( int i = 1; i <= number; i++ ) {
            result += i;
        }
        result *= result;
        return result;
    }

    int sum_of_squares(int number){
        int result{};
        for ( int i = 1; i <= number; i++ ) {
            result += i * i;
        }
        return result;
    }

    int difference(int number) {
        return square_of_sum(number) - sum_of_squares(number);
    }

}  // namespace difference_of_squares
